### ISSUE TITLE
ui5-dialog: Header text should be read out

### ISSUE THREAD

**Issue by elenastoyanovaa** [SAP/ui5-webcomponents#1280]
**Describe the bug**
The header text is not read out when the user opens a dialog.

**To reproduce**
1. Go to https://sap.github.io/ui5-webcomponents/playground/components/Dialog/
2. Run JAWS with Chrome on Window (should also be reproducible with VoiceOver)
3. Open the Dialog -> The Dialog header is not read out
4. Inspect the element -> There is no aria-labelledby attribute to the element with role=dialog.

**Expected behavior**
When a header/header text is provided it should be spoken (linked via aria-labelledby), when the user opens a dialog.
